src/main/java/com/takipi/oss/storage/resources/fs/multifetcher/MultiFetcher.java

```diff
@@ -14,33 +14,99 @@
 import java.util.List;
 
 public class MultiFetcher {
-    
+
+    private static final long DEFAULT_MAX_CACHE_SIZE = 33554432; // 32 MB
+
+    private static class PartSizeEstimator {
+
+        private static long MIN_LOADED_PARTS_FOR_SIZE_ESTIMATION = 10;
+        private static int DEFAULT_PART_SIZE_ESTIMATION = 1700;
+        private static long MAX_TOTAL_SIZE = 1L << 30;
+
+        private long totalSizeLoaded = 0;
+        private long numberOfPartsLoaded = 0;
+
+        synchronized void updateStats(long size) {
+            if (totalSizeLoaded < MAX_TOTAL_SIZE) {
```

Comment:
It is going to take more than 1000 years to reach the limit of long (assuming there is 100000 bytes per second).
No need for this check

Reply:
ok